### ISSUE TITLE
manage release/debug builds

### ISSUE THREAD

**Issue by celery6** [waterloo-rocketry/cansw_processor_canards#45]
joe once said,
> Also figure out once and for all/document how to  set up multiple build configs cause that was a nightmare to try and figure out last minute

**Comment by celery6** [comment]
adding to this: what compiler optimization if any should we use for production?

**Comment by Joe-Joe-Joe-Joe** [comment]
I don't think we are even close to pushing size limits. So probably `-O3`? I know we got scared doing it last minute last year. I don't know all the implications, but I think it just means testing has to happen with it on, people need to use `volatile`, etc etc
